### PR TITLE
gracefully shutdown master, pserver, fix gometalinter errors

### PR DESCRIPTION
**Pull request by helinwang**
Fixes: https://github.com/PaddlePaddle/Paddle/issues/2599

### CODE REVIEW

File: go/pserver/etcd_client.go [hunk 1]
```diff
@@ -165,26 +175,10 @@ func (e *EtcdClient) registerPserverEtcd(ctx context.Context, port int) (int, er
...
 
 			if ps == "" {
-				resp, err := e.etcdClient.Grant(context.TODO(), 5)
-				if err != nil {
-					log.Fatal(err)
-				}
 				// find the first id and write info
 				pserverAddr := e.externalIP + ":" + strconv.Itoa(port)
-				c.Put(psKey, pserverAddr, clientv3.WithLease(resp.ID))
+				c.Put(psKey, pserverAddr, clientv3.WithLease(e.sess.Lease()))
```
Comment (typhoonzero) [inline go/pserver/etcd_client.go]: 赞，使用session会简单很多～

File: go/master/etcd_client.go [hunk 2]
```diff
@@ -39,6 +39,7 @@ type EtcdClient struct {
 	statePath string
 	client    *clientv3.Client
 	lock      *concurrency.Mutex
+	sess      *concurrency.Session
 }
 
 // NewEtcdClient creates a new EtcdClient.
```
Comment (typhoonzero) [inline go/master/etcd_client.go]: Below "TODO" comments can be removed.

Comment (helinwang) [inline go/master/etcd_client.go]: Thanks! Done.

**Comment by helinwang** [comment]
@typhoonzero Thanks! Fixed.

**Pull request merged**